sas-viya/azuredeploy.json

```diff
@@ -0,0 +1,1658 @@
+              },
+              "primary": true,
+              "privateIPAddressVersion": "IPv4",
+              "applicationSecurityGroups": [
+                {
+                  "id": "[resourceId('Microsoft.Network/applicationSecurityGroups', 'github-accessor')]"
+                },
+                {
+                  "id": "[resourceId('Microsoft.Network/applicationSecurityGroups', 'sas-services-accessor')]"
+                },
+                {
+                  "id": "[resourceId('Microsoft.Network/applicationSecurityGroups', 'sas-viya-provider')]"
+                }
+              ]
+            }
+          }
+        ],
+        "dnsSettings": {
+          "dnsServers": [],
+          "appliedDnsServers": []
+        },
+        "enableAcceleratedNetworking": false,
+        "enableIPForwarding": false,
+        "networkSecurityGroup": {
+          "id": "[resourceId('Microsoft.Network/networkSecurityGroups', 'Viya_NetworkSecurityGroup')]"
+        },
+        "primary": true
+      },
+      "dependsOn": [
+        "[resourceId('Microsoft.Network/networkSecurityGroups', 'Viya_NetworkSecurityGroup')]",
+        "[resourceId('Microsoft.Network/applicationSecurityGroups', 'github-accessor')]",
+        "[resourceId('Microsoft.Network/applicationSecurityGroups', 'sas-services-accessor')]",
+        "[resourceId('Microsoft.Network/applicationSecurityGroups', 'sas-viya-provider')]",
+        "[resourceId('Microsoft.Resources/deployments', 'CreateNetwork')]"
+      ]
+    },
+    {
+      "comments": "Diagnostic storage group for the SAS Viya stack",
+      "type": "Microsoft.Storage/storageAccounts",
+      "sku": {
+        "name": "Standard_LRS",
+        "tier": "Standard"
+      },
+      "kind": "StorageV2",
+      "name": "[variables('DiagnosticStorageGroupName')]",
+      "apiVersion": "2018-02-01",
+      "location": "[parameters('location')]",
+      "tags": {},
+      "scale": null,
+      "properties": {
+        "networkAcls": {
+          "bypass": "AzureServices",
+          "virtualNetworkRules": [],
+          "ipRules": [],
+          "defaultAction": "Allow"
+        },
+        "supportsHttpsTrafficOnly": false,
+        "encryption": {
+          "services": {
+            "file": {
+              "enabled": true
+            },
+            "blob": {
+              "enabled": true
+            }
+          },
+          "keySource": "Microsoft.Storage"
+        }
+      },
+      "dependsOn": []
+    },
+    {
+      "comments": "The CAS controler for this environment.",
+      "type": "Microsoft.Compute/virtualMachines",
+      "name": "[variables('CASController_name')]",
+      "apiVersion": "2018-04-01",
+      "location": "[parameters('location')]",
+      "scale": null,
+      "properties": {
+        "hardwareProfile": {
+          "vmSize": "[json(split(reference('AnsiblePhase2GetCASSizingFromLicense').outputs.instanceView.value.statuses[0].message, '#DATA#')[1]).CAS_VM_SIZE]"
```

Comment:
Yes, depending on licensed core counts, it creates one of the following: Standard_E8s_v3, Standard_E16s_v3, or Standard_E32s_v3.

Reply:
What if I can't create E sizes due to policy, location, quota, capacity?

Reply:
The E series is supported in almost every region and is the recommended size. Since we have documented in the readme that these are the sizes that are being generated and there are only edge cases where it might fail, we would consider this an acceptable risk.